docs/contributor_guide/release.rst

```diff
@@ -0,0 +1,44 @@
+.. release_guide
+
+Release Process
+---------------
+
+This page outlines the process for creating a new Fairlearn release.
+The following steps assume git remote's `origin` points to
+`fairlearn/fairlearn` (in practical terms, that the work is being
+done on a clone of `fairlearn/fairlearn` and not on a fork).
+
+#. Create the branch for the release:
+
+    :code:`git checkout -b release/v<x.y.z>`
+
+#. Push the branch to GitHub:
+
+    :code:`git push -u origin release/v<x.y.z>`
+
+#. On the release branch, create a PR to:
+
+    #. Update the version in `__init__.py` to `x.y.z`
+    #. Update the version in the ReadMe
+
+#. Run the `release pipeline <https://dev.azure.com/responsibleai/fairlearn/_build?definitionId=60>`_
```

Comment:
This is the part which requires manual approval from you or me, right?

Reply:
I just saw you filed #757  .... we might well be the only ones who can even queue the release.